### PR TITLE
[FIX] website_crm_partner_assign: fixing broken urls in mail template

### PR DESCRIPTION
**Pull request by malv-odoo**
Issue:
-------------------
In the lead forward mail template, there are some broken links, when we click on those links they are appended to the current url and redirect to a non-expected page.

Steps to reproduce:
-------------------
1. Go to CRM -> Create a new opportunity
2. Click on this new opportunity -> Assigned Partner
3. Select an assigned partner and click “Send Email”
4. Go to Settings -> Technical -> Emails
5. When clicking on “Partner Portal” it redirects to a non-existing page

Cause:
-------------------
The issue is happening because the urls are badly formatted and the portal URL were getting the value from `record.get_base_url `but it is not working anymore and we had to pass the corrected value to the template 

OPW-3074049

### CODE REVIEW

File: addons/website_crm_partner_assign/data/mail_template_data.xml [hunk 1]
```diff
@@ -75,9 +75,9 @@
                 <tr><td valign="middle" align="left" style="opacity: 0.7;">
                     <t t-out="user.company_id.phone or ''">+1 650-123-4567</t>
                     <t t-if="user.company_id.phone and (user.company_id.email or user.company_id.website)">|</t>
-                    <a t-if="user.company_id.email" t-attf-href="'mailto:%s' % {{ user.company_id.email }}" style="text-decoration:none; color: #454748;" t-out="user.company_id.email or ''">info@yourcompany.com</a>
+                    <a t-if="user.company_id.email" t-att-href="'mailto:%s' % user.company_id.email" style="text-decoration:none; color: #454748;" t-out="user.company_id.email or ''">info@yourcompany.com</a>
```
Comment (nle-odoo) [inline addons/website_crm_partner_assign/data/mail_template_data.xml:78]: looks good to me, alternatively the t-attf could be kept with something like:

```diff
- 'mailto:%s' % {{ user.company_id.email }}
+ {{ 'mailto:%s' % user.company_id.email }}
```

File: addons/website_crm_partner_assign/data/mail_template_data.xml [hunk 2]
```diff
@@ -41,7 +41,7 @@
...
                             </ol>
                             <t t-if="ctx.get('partner_in_portal')">
-                                Please connect to your <a href="{{ '%s?db=%s' % (record.get_base_url(), record.env.cr.dbname) }}">Partner Portal</a> to get details. On each lead are two buttons on the top left corner that you should press after having contacted the lead: "I'm interested" &amp; "I'm not interested".<br/>
+                                Please connect to your  <a t-att-href="'%s/?db=%s' % (ctx['partner_id'].get_base_url(), ctx['partner_id'].env.cr.dbname)">Partner Portal</a> to get details. On each lead are two buttons on the top left corner that you should press after having contacted the lead: "I'm interested" &amp; "I'm not interested".<br/>
```
Comment (nle-odoo) [inline addons/website_crm_partner_assign/data/mail_template_data.xml:44]: alternatively we could just change the href to t-attf-href and not change the attribute value

**Review by nle-odoo — APPROVED** [review]
looks good to me

**Comment by tde-banana-odoo** [comment]
As it comes from https://github.com/odoo/odoo/commit/4813f429976a609513aa992a9813956165403872 isn't it broken in earlier versions ?

**Comment by malv-odoo** [comment]
> As it comes from [4813f42](https://github.com/odoo/odoo/commit/4813f429976a609513aa992a9813956165403872) isn't it broken in earlier versions ?

yep, you are right, the same issue is also present in 15.0. I just rebase the PR 🙂

**Comment by tde-banana-odoo** [comment]
Don't you have to export translations again (and therefore break them) ?

**Comment by malv-odoo** [comment]
@tde-banana-odoo 
In this case, We can do it in stable or we should retarget it to master?

File: addons/website_crm_partner_assign/data/mail_template_data.xml [hunk 3]
```diff
@@ -40,7 +40,7 @@
...
                             </ol>
                             <t t-if="ctx.get('partner_in_portal')">
-                                Please connect to your <a href="{{ record.get_portal_url() }}">Partner Portal</a> to get details. On each lead are two buttons on the top left corner that you should press after having contacted the lead: "I'm interested" &amp; "I'm not interested".<br/>
+                                Please connect to your <a t-att-href="'%s' % (ctx['partner_portal_url'])">Partner Portal</a> to get details. On each lead are two buttons on the top left corner that you should press after having contacted the lead: "I'm interested" &amp; "I'm not interested".<br/>
```
Comment (tde-banana-odoo) [inline addons/website_crm_partner_assign/data/mail_template_data.xml:43]: Relying on context instead of using "record" means your template is broken when not used in your specific method, as context will not be set. Why this change ? Does not seem necessary to me ?

**Comment by tde-banana-odoo** [comment]
@robodoo r+

Thanks for followup :) as it is indeed broken we can break translations.

**Comment by kitan191** [comment]
@robodoo retry

**Comment by nle-odoo** [comment]
robodoo r+ delegate=Ephoniz
